homeassistant/components/velbus/climate.py

```diff
@@ -79,3 +83,10 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set the new preset mode."""
         await self._channel.set_preset(PRESET_MODES[preset_mode])
         self.async_write_ha_state()
+
+    @api_call
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set the new hvac mode."""
+        if hvac_mode not in self.hvac_mode:
+            await self._channel.set_mode(hvac_mode)
+            self.async_write_ha_state()
```

Comment:
This makes no sense, please fix it

Reply:
This essentially says - if the current mode is not the same as the requested mode - switch to the requested mode.. if already in requested mode.. dont do anything..

Reply:
"not in" is not an idiomatic way to test for string unequality in Python.

I think you misunderstood my comment here: <https://github.com/home-assistant/core/pull/106570#discussion_r1627170542>

What I meant was that it's possible for the user to pass an `hvac_mode` which is not in `hvac_modes`. That should be detected by something like this:

```py
        if hvac_mode not in self.hvac_modes:
            raise ServiceValidationError(
                translation_domain=DOMAIN,
                translation_key=unsupported_hvac_mode
                translation_placeholders={"hvac_mode":hvac_mode}
```